event.go

```diff
@@ -60,3 +100,17 @@ func (e *Event) Cancel(err ...error) {
 func (e *Event) Async() {
 	e.async = true
 }
+
+// Transition can be called to transition to another state from a callback.
+// Currently, this only works in an enter state or after event callback.
+func (e *Event) Transition(ctx context.Context, event string, args ...interface{}) error {
```

Comment:
This is a highly debatable change. The gain isn't much but it required introducing the callback type into the Event struct. It should be removed again.

Reply:
I reverted all of this and redid the whole commit history.